inject-java/src/main/java/io/micronaut/annotation/processing/AnnotationProcessingOutputVisitor.java

```diff
@@ -96,7 +99,50 @@ public OutputStream visitClass(String classname, io.micronaut.inject.ast.Element
             nativeOriginatingElements = new Element[0];
         }
         javaFileObject = filer.createClassFile(classname, nativeOriginatingElements);
-        return javaFileObject.openOutputStream();
+        OutputStream os = javaFileObject.openOutputStream();
+        return unwrapFilterOutputStream(os);
+    }
+
+    private OutputStream unwrapFilterOutputStream(OutputStream os) {
+        // FilterOutputStream and JavacFiler$FilerOutputStream is always using write(int) and killing performance, unwrap if possible
+        if (os instanceof FilterOutputStream) {
+            try {
+                Field outField = ReflectionUtils.getRequiredField(FilterOutputStream.class, "out");
+                outField.setAccessible(true);
```

Comment:
I think this would be called for each file we create. Could refactor this so we only do the reflection bit once

Reply:
Done